### PR TITLE
Some review remarks.

### PR DESCRIPTION
**Pull request by Gerrit91**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
+Along the way of implementing [*cluster-api*](https://github.com/kubernetes-sigs/cluster-api) on top of [*metal-stack*](https://github.com/metal-stack) we learnt something about [*kubebuilder*](https://github.com/kubernetes-sigs/kubebuilder)  which enables us to write reconciliation logic easily and we want to share that knowledge with you, so we built this project, an extremely simplified version of cluster which contains *metal-stack* resources. We will assume you already went through [*kubebuilder book*](https://book.kubebuilder.io) and are looking for more hands-on examples. By referencing the code in this project, you will be able to create a *CustomResourceDefinition* (CRD), write its reconciliation logic and deploy it.
```
Comment (Gerrit91) [inline README.md]:
```suggestion
Along the way of implementing [*cluster-api*](https://github.com/kubernetes-sigs/cluster-api) on top of [*metal-stack*](https://github.com/metal-stack) we learnt something about [*kubebuilder*](https://github.com/kubernetes-sigs/kubebuilder) which enables us to write reconciliation logic easily and we want to share that knowledge with you, so we built this project, an extremely simplified version of cluster which contains *metal-stack* resources. We will assume you already went through [*kubebuilder book*](https://book.kubebuilder.io) and are looking for more hands-on examples. By referencing the code in this project, you will be able to create a *CustomResourceDefinition* (CRD), write its reconciliation logic and deploy it.
```

File: README.md [hunk 2]
```diff
@@ -1,14 +1,21 @@
...
 
 ## Architecture
 
-We created two *CustomResourceDefinition* (CRD), `XCluster` and `XFirewall`, as shown in the following figure. `XCluster` represents the computer cluster which contains *metal-stack network* and `XFirewall`. `XFirewall` corresponds to *metal-stack firewall*. The circular arrows imply the nature of recociliation.
+We created two *CRD*, `XCluster` and `XFirewall`, as shown in the following figure. `XCluster` represents a cluster which contains *metal-stack network* and `XFirewall`. `XFirewall` corresponds to *metal-stack firewall*. The circular arrows imply the nature of reconciliation and also the corresponding [*controllers*](https://github.com/LimKianAn/xcluster/tree/main/controllers) which reconcile the states of the resources.
```
Comment (Gerrit91) [inline README.md]:
```suggestion
We created two *CRDs*, `XCluster` and `XFirewall`, as shown in the following figure. `XCluster` represents a cluster which contains *metal-stack network* and `XFirewall`. `XFirewall` corresponds to *metal-stack firewall*. The circular arrows imply the nature of reconciliation and also the corresponding [*controllers*](https://github.com/LimKianAn/xcluster/tree/main/controllers) which reconcile the states of the resources.
```

File: README.md [hunk 3]
```diff
@@ -18,19 +25,25 @@ Clone the repo of [*mini-lab*](https://github.com/metal-stack/mini-lab) and *xcl
...
 
 ```bash
 cd mini-lab
 make
 ```
 
-It's going to take some time to finish. From time to tiem, do
+It's going to take some time to finish. Behind the scene, a [kind](https://github.com/kubernetes-sigs/kind/) cluster is created, *metal-api* related kubernetes resources are deployed, and multiple *linux kernel-based virtual machines* are created for *metal-stack* switches and machines.
+
+From time to time, do
```
Comment (Gerrit91) [inline README.md:33]: "From time to time, do ... until two waiting machines are shown" --> add console output how this looks like

Comment (LimKianAn) [reply README.md:33]: It's already there. I will clean up the comments later.
```console
ID                                          LAST EVENT   WHEN     AGE  HOSTNAME  PROJECT  SIZE          IMAGE  PARTITION
e0ab02d2-27cd-5a5e-8efc-080ba80cf258        Waiting      8s                               v1-small-x86         vagrant
2294c949-88f6-5390-8154-fa53d93a3313        Waiting      8s                               v1-small-x86         vagrant
```

**Review by LimKianAn — APPROVED** [review]

**Pull request merged**